test/integration/integration_test.go

```diff
@@ -217,28 +218,60 @@ func ensureTestDirEmpty() {
 }
 
 //  To get values of nested keys, use following dot formating in Scenarios: key.nestedKey
-//  If an array is expected, use following formating: [value1 value2 value3].
-func configContains(configPath, expectingResult, expectedKeyPath, expectedValue string) error {
+//  If an array is expected, then expect: "[value1 value2 value3]"
+//  If empty string, non existing value are expected, then expect "<nil>"
+func getConfigValue(configPath string, keyPath string) (string, error) {
+	var keyValue string
 	data, err := ioutil.ReadFile(testDir + "/" + configPath)
 	if err != nil {
-		return fmt.Errorf("Cannot read config file: %s", err)
+		return "", fmt.Errorf("Cannot read config file: %v", err)
 	}
 	var values map[string]interface{}
 	json.Unmarshal(data, &values)
-	actualValue := ""
-	keyPath := strings.Split(expectedKeyPath, ".")
-	for _, element := range keyPath {
+	keyPathArray := strings.Split(keyPath, ".")
+	for _, element := range keyPathArray {
 		switch value := values[element].(type) {
 		case map[string]interface{}:
 			values = value
 		case []interface{}, nil, string, float64:
-			actualValue = fmt.Sprintf("%v", value)
+			keyValue = fmt.Sprintf("%v", value)
 		default:
-			return errors.New("Unexpected type in JSON config, not supported by testsuite")
+			return "", errors.New("Unexpected type in JSON config, not supported by testsuite")
 		}
 	}
-	if (expectingResult == "contains") && (actualValue != expectedValue) {
-		return fmt.Errorf("For key '%s' config contains unexpected value '%s'", expectedKeyPath, actualValue)
+	return keyValue, nil
+}
+
+func jsonConfigContains(configPath string, condition string, keyPath string, expectedValue string) error {
```

Comment:
jsonConfigContains == configValueMatches?

jsonConfigContains as name creates the image in my mind that we do a regexp over the whole file. However, the regexp is applied to the the value of a specific key.